### PR TITLE
[Merged by Bors] - feat: add `Equiv.prodSubtypeFstEquivSubtypeProd`

### PR DESCRIPTION
**Pull request by xroblot**
See [Zulip](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/A.20basic.20equivalence/near/437968727)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Equiv/Basic.lean [hunk 1]
```diff
@@ -1380,6 +1380,14 @@ def subtypeProdEquivProd {p : α → Prop} {q : β → Prop} :
   right_inv := fun ⟨⟨_, _⟩, ⟨_, _⟩⟩ => rfl
 #align equiv.subtype_prod_equiv_prod Equiv.subtypeProdEquivProd
 
+/-- A subtype of a `Prod` that depends only on the first component is equivalent to the
+corresponding subtype of the first type times the second type. -/
+def prodSubtypeEquivSubtypeProd {p : α → Prop} : {s : α × β // p s.1} ≃ {a // p a} × β where
```
Comment (mattrobball) [inline Mathlib/Logic/Equiv/Basic.lean:1385]:
```suggestion
def prodSubtypeFstEquivSubtypeProd {p : α → Prop} : {s : α × β // p s.1} ≃ {a // p a} × β where
```
To disambiguate?

Comment (xroblot) [reply Mathlib/Logic/Equiv/Basic.lean]: Sure. Sounds good.

**Comment by mattrobball** [comment]
Thanks!

bors delegate+

**Pull request closed without merging**